Assets/MixedRealityToolkit/Utilities/Editor/ScreenshotUtility.cs

```diff
@@ -0,0 +1,152 @@
+
+                return true;
+            }
+
+            // Make sure we have a valid camera to render from.
+            if (camera == null)
+            {
+                camera = Camera.main;
+
+                if (camera == null)
+                {
+                    Debug.Log("Failed to acquire a valid camera to capture a screenshot from.");
+
+                    return false;
+                }
+            }
+
+            // Create a camera clone with a transparent clear color.
+            var renderCamera = new GameObject().AddComponent<Camera>();
+            renderCamera.transform.position = camera.transform.position;
+            renderCamera.transform.rotation = camera.transform.rotation;
+            renderCamera.clearFlags = transparentClearColor ? CameraClearFlags.Color : camera.clearFlags;
+            renderCamera.backgroundColor = transparentClearColor ? new Color(0.0f, 0.0f, 0.0f, 0.0f) : camera.backgroundColor;
+            renderCamera.fieldOfView = camera.fieldOfView;
+            renderCamera.nearClipPlane = camera.nearClipPlane;
+            renderCamera.farClipPlane = camera.farClipPlane;
+
+            // Create a render texture for the camera clone to render into.
+            var width = Screen.width * superSize;
+            var height = Screen.height * superSize;
+            var renderTexture = new RenderTexture(width, height, 24, RenderTextureFormat.ARGB32);
+            renderTexture.antiAliasing = 8;
+            renderCamera.targetTexture = renderTexture;
+
+            // Render from the camera clone.
+            renderCamera.Render();
+
+            // Copy the render from the camera and save it to disk.
+            var outputTexture = new Texture2D(width, height, TextureFormat.ARGB32, false);
+            RenderTexture previousRenderTexture = RenderTexture.active;
+            RenderTexture.active = renderTexture;
+            outputTexture.ReadPixels(new Rect(0.0f, 0.0f, width, height), 0, 0);
+            outputTexture.Apply();
+            RenderTexture.active = previousRenderTexture;
+
+            try
+            {
+                File.WriteAllBytes(path, outputTexture.EncodeToPNG());
+            }
+            catch (Exception e)
+            {
+                Debug.LogException(e);
+
+                return false;
+            }
+            finally
+            {
+                UnityEngine.Object.DestroyImmediate(outputTexture);
+                UnityEngine.Object.DestroyImmediate(renderCamera.gameObject);
+                UnityEngine.Object.DestroyImmediate(renderTexture);
+            }
+
+            Debug.LogFormat("Screenshot captured to: {0}", path);
+
+            return true;
+        }
+
+        /// <summary>
+        /// Gets a screenshot file name based on time and located on the current user's desktop.
+        /// </summary>
+        /// <returns>The filename of the screenshot.</returns>
+        public static string GetScreenshotFileName()
```

Comment:
It would be really nice to pull this from `MixedRealityPreferences` so we could set the output path and screenshot format in the project settings.

Reply:
Oh I like that idea, I can add that!

Reply:
I just started going down this path but decided against it to keep this functionality independent of the MRTK (in case someone want's to just drop this script into older MRTK versions, or projects without the MRTK).